### ISSUE TITLE
In TF 0.13+ all list elements must have the same type

### ISSUE THREAD

**Issue by apeabody** [terraform-google-modules/terraform-google-gcloud#123]
### TL;DR

Historically `list(any)` was used for https://github.com/terraform-google-modules/terraform-google-gcloud/blob/6789fa726f443e36a3159f452eec0fc5f9a9ff4b/variables.tf#L35  However subsequent to https://github.com/hashicorp/terraform/issues/26265, all list elements must have the same type.  So for example this fails: `module_depends_on = [time_sleep.wait_300_seconds, null_resource.test]` The given value is not suitable as all list elements must have the same type.

@bharathkkb - When you have time, could you confirm/feedback as if so this might apply to several modules? The vast majority of `module_depends_on` only have a single argument, are therefore work as expected.  However we might want to switch to a string or update the documentation to clarify the situation.

### Terraform Configuration

```hcl
N/A
```

### Terraform Version

```sh
Terraform v1.2.2
```

**Comment by bharathkkb** [comment]
@apeabody updating docs to clarify this sgtm. IIUC end users can also workaround this by type coercion if needed.

**Comment by maidmaid** [comment]
A workaround could be to use an intermediate null_resource depending on the real resources:

```terraform
resource "null_resource" "depends_on_sleep_and_test" {
  depends_on = [time_sleep.wait_300_seconds, null_resource.test]
}

module "execute_create_admin" {
  source            = "terraform-google-modules/gcloud/google"
  module_depends_on = [null_resource.depends_on_sleep_and_test]
}
```
